### PR TITLE
Sdbkv refactor

### PR DESCRIPTION
**Pull request by ret2libc**
~~This PR is based on https://github.com/radare/sdb/pull/158 .~~

~~The only difference is the last commits.~~ The commit introduces helper macros in the sdbht.h file to access SdbKv->key/value/key_len/value_len since they are fields of the "base structure" HtKv. In another PR i'm going to change those fields, so I want to introduce these helpers to make those change easier.

### CODE REVIEW

**Comment by ret2libc** [comment]
I intend to sync radare2 after merging this commit, so that in next PRs I can change
```
struct SdbKv {
	//sub of HtKv so we can cast safely
	char *key;
	char *value;
	ut32 key_len;
	ut32 value_len;
	ut32 cas;
	ut64 expire;
};
```
to
```
typedef struct sdb_kv {
	//sub of HtKv so we can cast safely
	HtKv base;
	ut32 cas;
	ut64 expire;
} SdbKv;
```
easily, without rewriting everything. Also, this will be helpful because key/value are both `char*` in SdbKv, but not in HtKv, so a cast will be needed.

File: src/sdbht.h [hunk 1]
```diff
@@ -14,6 +14,11 @@ typedef struct sdb_kv {
 	ut64 expire;
 } SdbKv;
 
+#define SDBKV_KEY(kv) ((kv)->key)
```
Comment (ret2libc) [inline src/sdbht.h]: @radare Do you prefer `static inline` functions instead of these macros?

File: src/query.c [hunk 2]
```diff
@@ -335,8 +335,7 @@ SDB_API char *sdb_querys (Sdb *r, char *buf, size_t len, const char *_cmd) {
 			SdbListIter *iter;
 			SdbKv *kv;
 			ls_foreach (list, iter, kv) {
-				//eprintf ("(%s)(%s)\n", kv->key, kv->value);
-				foreach_list_cb (&user, kv->key, kv->value);
+				foreach_list_cb (&user, SDBKV_KEY (kv), SDBKV_VALUE (kv));
```
Comment (radare) [inline src/query.c]: why this change i dont see a reason to have this macro anymore

**Comment by ret2libc** [comment]
> We probably dont need those macros at all

See this comment https://github.com/radare/sdb/pull/162#issuecomment-426960357

File: src/query.c (hunk 2 above)
Comment (radare) [reply src/query.c]: i understand the concern of having macros to make this easier to refactor, but this is making the code harder to read, what about just having 1 letter macros for that? K(kv) and V(kv)
